src/mvAppItem.cpp

```diff
@@ -2660,32 +2699,35 @@ DearPyGui::GetEntityParser(mvAppItemType type)
         args.push_back({ mvPyDataType::Bool, "equal_aspects", mvArgType::KEYWORD_ARG, "False", "primary x and y axes will be constrained to have the same units/pixel (does not apply to auxiliary y-axes)"});
+        args.push_back({ mvPyDataType::Bool, "no_legend", mvArgType::KEYWORD_ARG, "False", "the legend will not be displayed"});
+        args.push_back({ mvPyDataType::Bool, "no_inputs", mvArgType::KEYWORD_ARG, "False", "the user will not be able to interact with the plot"});
+        args.push_back({ mvPyDataType::Bool, "no_frame", mvArgType::KEYWORD_ARG, "False", "the ImGui frame will not be rendered"});
+        // args.push_back({ mvPyDataType::Bool, "canvas_only", mvArgType::KEYWORD_ARG, "False", "shortcut for: no_title + no_legend + no_menu + no_box_select + no_mouse_text"});
         args.push_back({ mvPyDataType::Bool, "use_local_time", mvArgType::KEYWORD_ARG, "False", "axis labels will be formatted for your timezone when" });
         args.push_back({ mvPyDataType::Bool, "use_ISO8601", mvArgType::KEYWORD_ARG, "False", "dates will be formatted according to ISO 8601 where applicable (e.g. YYYY-MM-DD, YYYY-MM, --MM-DD, etc.)" });
         args.push_back({ mvPyDataType::Bool, "use_24hour_clock", mvArgType::KEYWORD_ARG, "False", "times will be formatted using a 24 hour clock" });
+        args.push_back({ mvPyDataType::Bool, "delete_rect", mvArgType::KEYWORD_ARG, "True", "allows to delete drag rect with double left mouse click" });
 
         // key modifiers
-        args.push_back({ mvPyDataType::Integer, "pan_button", mvArgType::KEYWORD_ARG, "internal_dpg.mvMouseButton_Left", "enables panning when held" });
-        args.push_back({ mvPyDataType::Integer, "pan_mod", mvArgType::KEYWORD_ARG, "-1", "optional modifier that must be held for panning" });
-        args.push_back({ mvPyDataType::Integer, "fit_button", mvArgType::KEYWORD_ARG, "internal_dpg.mvMouseButton_Left", "fits visible data when double clicked" });
-        args.push_back({ mvPyDataType::Integer, "context_menu_button", mvArgType::KEYWORD_ARG, "internal_dpg.mvMouseButton_Right", "opens plot context menu (if enabled) when clicked" });
-        args.push_back({ mvPyDataType::Integer, "box_select_button", mvArgType::KEYWORD_ARG, "internal_dpg.mvMouseButton_Right", "begins box selection when pressed and confirms selection when released" });
-        args.push_back({ mvPyDataType::Integer, "box_select_mod", mvArgType::KEYWORD_ARG, "-1", "begins box selection when pressed and confirms selection when released" });
-        args.push_back({ mvPyDataType::Integer, "box_select_cancel_button", mvArgType::KEYWORD_ARG, "internal_dpg.mvMouseButton_Left", "cancels active box selection when pressed" });
-        args.push_back({ mvPyDataType::Integer, "query_button", mvArgType::KEYWORD_ARG, "internal_dpg.mvMouseButton_Middle", "begins query selection when pressed and end query selection when released" });
```

Comment:
`query_button`, `query_mod`, `query_toggle_mod`:
It took me a great deal of research to understand what happened to the query feature. Here are a couple of observations.

First of all, the old way to draw query rects was way more flexible in terms of specifying the button and the modifiers. The updated version suggests a hardcoded "click one mouse button while holding the other" method, which, being itself an interesting gesture, is not very intuitive. Moreover, for some users it might even be difficult to perform, and might cause the mouse to move slightly and affect the end result.

I do not see an easy way to preserve `query_button` and `query_mod`, so we'll have to say good-bye to these two. I wonder though whether we should add them back to the parser - with the `DEPRECATED_REMOVE_KEYWORD_ARG` flag. This way the existing scripts won't crash but will instead spit warnings, and might be able to work normally except for plot queries.

For `query_toggle_mod`, I think we _can_ retain that. I'm going to add a patch that implements it. Unfortunately it uses kind of a hack on ImPlot to prevent it from messing up the query rect; however, the current solution with `SelectCancel` click is actually a hack as well - it only works because ImPlot processes `SelectCancel` on **mouse up**, while mvPlot does that on **mouse down** (i.e. earlier). I'm not blaming anyone (especially since that `SelectCancel` code came from ImPlot demo), just trying to say that we can easily exchange one hack to another.

Another problem is that all the standard modifiers (Ctrl, Alt, and Shift) are already used by default in ImPlot, with Ctrl now assigned to `OverrideMod`. I honestly think that ImPlot could do a better job on handling `OverrideMod`, e.g. check it on mouse down, so that the same modifier could be pressed _during selection_ to change the semantics. In my patch, I'll be hacking `OverrideMod` on the fly to achieve that.

In future, I hope ImPlot provides a better API to handle user input for queries. I've opened an issue in ImPlot repo to address that, see https://github.com/epezent/implot/issues/566.

Reply:
Here's a branch (in my fork of DPG) that implements `query_toggle_mod` on the updated ImPlot.
https://github.com/v-ein/DearPyGui-fixes/compare/porting...v-ein:DearPyGui-fixes:feature/query-toggle-mod?expand=1

You can probably cherry-pick commits or at least copy-paste the changes.